From e555416a60246ded989d7f301e0dd59d82d76476 Mon Sep 17 00:00:00 2001
From: Cong Jiajia <jcong@codeaurora.org>
Date: Thu, 11 May 2017 18:26:13 +0800
Subject: [PATCH 05/13] libstagefright: encoder must exist when source starting

when trying to start source, need to make sure encoder still exist.
Although encoder starts before source, it may encounter error and be
released before starting source, this will cause camera hardware can't
be released.

Change-Id: I5d5d474c5479001ec9d1d97a67c268adbf153337
---
 media/libstagefright/MediaCodecSource.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/MediaCodecSource.cpp b/media/libstagefright/MediaCodecSource.cpp
index 20881a4bb..58fcc43e4 100644
--- a/media/libstagefright/MediaCodecSource.cpp
+++ b/media/libstagefright/MediaCodecSource.cpp
@@ -761,8 +761,8 @@ status_t MediaCodecSource::feedEncoderInputBuffers() {
 }
 
 status_t MediaCodecSource::onStart(MetaData *params) {
-    if (mStopping) {
-        ALOGE("Failed to start while we're stopping");
+    if (mStopping || mOutput.lock()->mEncoderReachedEOS) {
+        ALOGE("Failed to start while we're stopping or encoder already stopped due to EOS error");
         return INVALID_OPERATION;
     }
     int64_t startTimeUs;
-- 
2.17.1

